Sources/ArcGISToolkit/Extensions/Viewpoint.swift

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+
+public extension Viewpoint {
+    /// Creates a new viewpoint with the same target geometry and scale but with a new rotation.
+    /// - Parameter rotation: The rotation for the new viewpoint.
+    /// - Returns: A new viewpoint.
+    func withRotation(_ rotation: Double) -> Viewpoint {
+        switch self.kind {
+        case .centerAndScale:
+            return Viewpoint(
+                center: self.targetGeometry as! Point,
+                scale: self.targetScale,
+                rotation: rotation
+            )
+        case.boundingGeometry:
+            return Viewpoint(
+                boundingGeometry: self.targetGeometry,
+                rotation: rotation
+            )
+        }
+    }
+}
```

Comment:
This strikes me as the type of thing we should add a test for, what do you think? I prototyped two quick tests that should hit both cases:

```swift
import ArcGIS
import XCTest

final class ViewpointTests: XCTestCase {
    func testWithRotationWithBoundingGeometry() {
        let rotatedViewpoint = Viewpoint(boundingGeometry: Polyline.simple.extent, rotation: -45)
        XCTAssertEqual(rotatedViewpoint.rotation, -45)
        
        let nonRotatedViewpoint = rotatedViewpoint.withRotation(.zero)
        XCTAssertEqual(nonRotatedViewpoint.rotation, 0)
    }
    
    func testWithRotationWithCenterAndScale() {
        let rotatedViewpoint = Viewpoint(center: .esriRedlands, scale: 1_000, rotation: 45)
        XCTAssertEqual(rotatedViewpoint.rotation, 45)
        
        let nonRotatedViewpoint = rotatedViewpoint.withRotation(.zero)
        XCTAssertEqual(nonRotatedViewpoint.rotation, 0)
    }
}

private extension Point {
    static var esriRedlands: Point {
        .init(
            x: -117.19494,
            y: 34.05723,
            spatialReference: .wgs84
        )
    }
}

private extension Polyline {
    static var simple: Polyline {
        .init(points: [
            Point(x: -117, y: 34, spatialReference: .wgs84),
            Point(x: -116, y: 34, spatialReference: .wgs84),
            Point(x: -116, y: 33, spatialReference: .wgs84)
        ])
    }
}
```

Reply:
Yes, those look good, thanks for the suggestion!